CONTRIBUTING.md

```diff
@@ -1,6 +1,94 @@
 
-When preparing a contribution we ask that you read the developers guide (see `dev_guide/`) to check for design and coding style intentions.
-As described in the build tools documentation (`build_tools/build_system_readme.md`) the `flint` program is used to apply linting rules to the ASiMoV-CCS source.
+When preparing a contribution we ask that you read the developers guide (see `dev_guide/`) to check
+for design and coding style intentions.
+As described in the build tools documentation (`build_tools/build_system_readme.md`) the `flint`
+program is used to apply linting rules to the ASiMoV-CCS source.
 
-Before a contribution is merged it will be reviewed by one of the ASiMoV-CCS developers, in particular to check the coding standard is followed and to verify the full test suite must still pass.
+Before a contribution is merged it will be reviewed by one of the ASiMoV-CCS developers, in
+particular to check the coding standard is followed and to verify the full test suite still
+passes.
+
+# Instructions for contributing on GitHub
+
+If you have not done so already, create a fork of the ASiMoV-CCS repository where you can prepare
+your contributions.
+
+The `develop` branch contains the latest version of the code, new contributions should therefore
+start from this branch.
+
+## Before starting a contribution
+
+We appreciate people taking the time and effort to contribute to ASiMoV-CCS.
+To prevent potential duplication of effort, we ask that before embarking on making code changes that
+you open an issue on the GitHub repository outlining the proposed change and why it is necessary, it
+may be that someone is already working on something similar or that there would be a conflict with
+some other planned developments, this will give the opportunity to flag potential issues in advance.
+
+## Create a new branch
+
+Hopefully there are no clashes with the proposed contribution and ongoing work and the proposed work
+can proceed.
+In the issue proposing the contribution click the `Create a branch` link, this will open a popup
+where you can name your branch and select your forked repository as the branch destination.
+At this point you will be able to checkout your branch locally and begin working.
+
+### Before you start
+
+If you have already worked on CCS then you should ensure your local repository view of `develop` is
+up to date before beginning new work.
+To do so we recommend using `git rebase` to keep the git history clean, assuming that the main
+ASiMoV-CCS repository is `origin` and your branch is called `my-branch`:
+```
+git checkout develop
+git pull origin/develop
+git checkout my-branch
+git rebase develop
+```
+
+You are now ready to begin.
+
+### Keeping up to date
+
+We recommend periodically pulling the `origin/develop` branch and merging into your working branch
+if the development of your contribution continues for some time, this will ensure that it does not
+stray too far from the eventual merge target.
```

Comment:
I think this should also be a rebase instead of a merge.

Reply:
Done